lib/streamlit/runtime/caching/memo_decorator.py

```diff
@@ -353,7 +358,7 @@ def wrapper(f):
                     show_spinner=show_spinner,
                     suppress_st_warning=suppress_st_warning,
                     max_entries=max_entries,
-                    ttl=ttl,
+                    ttl=cast(float, ttl),
```

Comment:
we cast `ttl` to `float` to make mypy happy

Reply:
Small nit: let's introduce a new param instead of mutating `ttl`, with the intention of avoiding the cast:

```python
ttl_seconds: float
if is_instance(ttl, timedelta):
    ttl_seconds = ttl.total_seconds()
else:
    ttl_seconds = ttl

def wrapper(f):
    # blah blah
    ttl = ttl_seconds
    ...
```